src/spikeinterface/curation/auto_merge.py

```diff
@@ -356,6 +362,12 @@ def compute_merge_unit_groups(
             )
             outs["pairs_decreased_score"] = pairs_decreased_score
 
+        elif step == "slay_score":
+
+            M_ij = compute_slay_matrix(sorting_analyzer, params["k1"], params["k2"], params["template_diff_thresh"])
```

Comment:
to save time, we should compute this only on the `pair_mask` that survived the template similarity step!

This could be an optional input of the function: if given, all pairs with mask False will be set to 0, so we keep the correct shape